src/main/scala/org/phenoscape/kb/PresenceAbsenceOfStructure.scala

```diff
@@ -159,20 +163,39 @@ object PresenceAbsenceOfStructure {
 
   private def unOBO(uri: String): String = uri.replaceAllLiterally("http://purl.obolibrary.org/obo/", "")
 
-  def buildMatrixQuery(entityClass: OWLClassExpression, taxonClass: OWLClassExpression): Query =
-    construct(t('taxon, 'relation, 'entity),
-              t('taxon, rdfsLabel, 'taxon_label),
-              t('entity, rdfsLabel, 'entity_label)) from "http://kb.phenoscape.org/" where (bgp(
-      t('taxon, 'relation, 'entity),
-      t('taxon, rdfsLabel, 'taxon_label),
-      t('entity, rdfsLabel, 'entity_label),
-      t('entity, rdfsSubClassOf, entityClass.asOMN),
-      t('taxon, rdfsSubClassOf, taxonClass.asOMN)
-    ),
-    new ElementFilter(
-      new E_OneOf(new ExprVar('relation),
-                  new ExprList(
-                    List[Expr](new NodeValueNode(has_presence_of), new NodeValueNode(has_absence_of)).asJava))))
+  def buildMatrixQuery(entityClass: OWLClassExpression,
+                       taxonClass: OWLClassExpression,
+                       includeSubClasses: Boolean): Query = {
+    val subClasses =
+      sparql"""
+                    ?entity $rdfsSubClassOf ${entityClass.asOMN} .
+                    ?taxon $rdfsSubClassOf ${taxonClass.asOMN}
+              """
+    val withoutSubClasses =
+      sparql"""
+               ?taxon ?relation ?entity .
+                ?taxon ${KBVocab.rdfsLabel} ?taxon_label .
+                ?entity ${KBVocab.rdfsLabel} ?entity_label .
+                VALUES ?entity {${entityClass.asOMN}}
+                VALUES ?taxon {${taxonClass.asOMN}}
```

Comment:
To just put the terms in it should just be `$entityClass` and `$taxonClass`, not `.asOMN`.